src/managedwriter/stream_connection.ts

```diff
@@ -238,13 +253,38 @@ export class StreamConnection extends EventEmitter {
     return this._streamId;
   };
 
+  private hasPendingWrites(): boolean {
+    return this._pendingWrites.length > 0;
+  }
+
   private getNextPendingWrite(): PendingWrite | null {
     if (this._pendingWrites.length > 0) {
-      return this._pendingWrites[0];
+      return this._pendingWrites[this._pendingWrites.length - 1];
     }
     return null;
   }
 
+  private resendAllPendingWrites() {
+    const pendingWrites = [...this._pendingWrites]; // copy array;
```

Comment:
this is needed because now when resending the requests, it is added to the `pendingWrites` queue right away instead of waiting for the `write` call to finish. This was causing a infinite loop as the queue was getting items removed (with pop() call) and added right away, so the queue was never empty and the while loop was never ending. 

so we create a copy of the queue, resend requests based on the copy and also empty the internal queue at the same time